tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/TestRewriterWithOOPSLA19.scala

```diff
@@ -17,6 +18,11 @@ class TestRewriterWithOOPSLA19
     with TestSymbStateRewriterTlc with TestSymbStateRewriterTuple with TestPropositionalOracle with TestSparseOracle
     with TestUninterpretedConstOracle {
   override protected def withFixture(test: OneArgTest): Outcome = {
-    test(oopsla19RewriterType)
+    solverContext = new PreproSolverContext(new Z3SolverContext(SolverConfig.default.copy(debug = true,
+                smtEncoding = oopsla19EncodingType)))
+    arena = Arena.create(solverContext)
+    val result = test(oopsla19EncodingType)
+    solverContext.dispose()
+    result
```

Comment:
This code will probably appear in many tests. Does it make sense to refactor this method into a trait?

Reply:
I don't think so, every class with `JUnitRunner` has a similar, yet different, setup. See `TestArenaWithOOPSLA19`, `TestSeqModelCheckerWithOOPSLA19`, `TestRecordingSolverContextWithOOPSLA19`, and `TestTransitionExecutorWithOfflineAndOOPSLA19`.